Lib/test/clinic.test

```diff
@@ -3379,26 +3381,27 @@ test_vararg(PyObject *module, PyObject *const *args, Py_ssize_t nargs, PyObject
     static const char * const _keywords[] = {"a", NULL};
     static _PyArg_Parser _parser = {NULL, _keywords, "test_vararg", 0};
     PyObject *argsbuf[2];
+    Py_ssize_t varargssize = Py_MAX(nargs - 1, 0);
     Py_ssize_t noptargs = 0 + (kwnames ? PyTuple_GET_SIZE(kwnames) : 0) - 1;
     PyObject *a;
-    PyObject *__clinic_args = NULL;
+    PyObject *const *__clinic_args = NULL;
 
-    args = _PyArg_UnpackKeywordsWithVararg(args, nargs, NULL, kwnames, &_parser, 1, 1, 0, 1, argsbuf);
+    args = _PyArg_UnpackKeywordsWithVarargFast(args, nargs, NULL, kwnames, &_parser, 1, 1, 0, 1, varargssize, argsbuf);
     if (!args) {
         goto exit;
     }
     a = args[0];
-    __clinic_args = args[1];
-    return_value = test_vararg_impl(module, a, __clinic_args);
+    __clinic_args = (PyObject *const *)args[1];
```

Comment:
It would be nice to avoid such unsafe casts. Maybe passing a `PyObject ***` into `_PyArg_UnpackKeywordsWithVarargFast` instead?

Reply:
This copy and cast are unnecessary.
Maybe `__clinic_args = args + 1;` is enough, note the `args` refers to the 2nd argument of function test_vararg, assuming it's not assigned in line 3389.